src/Passes/docs/src/UserGuide/IntroductionToProfiles.md

```diff
@@ -0,0 +1,89 @@
+# Introduction to profiles
+
+In this document we discuss QIR profiles. A QIR profile describes a subset of the generic QIR functionality and conventions. It
+is anticipated that most usages of the QIR specification will need to only use a subset of it. These subsets
+may further be subject to constraints such as how one allocate or acquire a qubit handle. We refer to such a subset with
+constraints as a profile. For instance, it is likely that early versions of quantum hardware will have a limited
+set of classical instructions available. With this in mind, the vendor or user of said hardware would define a profile
+that only contains a specified subset. One example of such a profile is the base profile,
+which only allows function calls and branching, but no arithmetic, classical memory, or classical registers.
+
+The generation of QIR according to the spec with no constraints would typically be performed by the frontend. A couple
+of examples are Q# or OpenQASM. However, for the generated QIR to be practical it is necessary to reduce it using a profile
```

Comment:
Similarly here, OpenQASM should be qualified as 2.0, 3.0, or both.

Reply:
Added